components/object_store/src/obkv/mod.rs

```
@@ -0,0 +1,940 @@
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        debug!(
+            "ObkvObjectStore abort_multipart operation, location:{location}, table_name:{table_name}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+        Ok(())
+    }
+
+    async fn get(&self, location: &Path) -> Result<GetResult> {
+        let instant = Instant::now();
+        let result = self.get_internal(location).await;
+
+        debug!(
+            "ObkvObjectStore get operation, location:{location}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+        result.box_err().map_err(|source| StoreError::NotFound {
+            path: location.to_string(),
+            source,
+        })
+    }
+
+    async fn get_range(&self, location: &Path, range: Range<usize>) -> Result<Bytes> {
+        let instant = Instant::now();
+
+        let table_name = self.pick_shard_table(location);
+        let meta =
+            self.read_meta(location)
+                .await
+                .box_err()
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source,
+                })?;
+
+        let mut range_buffer = Vec::with_capacity(range.end - range.start);
+        let covered_parts = meta
+            .compute_covered_parts(range)
+            .box_err()
+            .map_err(|source| StoreError::NotFound {
+                path: location.to_string(),
+                source,
+            })?;
+
+        for (index, key) in covered_parts.part_keys.iter().enumerate() {
+            let values = self
+                .client
+                .get(table_name, key.as_bytes())
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source: Box::new(source),
+                })?;
+
+            if let Some(bytes) = values {
```

Shall throw an error if `values` is None?

---

Done